src/ComposerRequireChecker/FileLocator/LocateComposerPackageDirectDependenciesSourceFiles.php

```diff
@@ -30,7 +50,7 @@ public function __invoke(string $composerJsonPath): Generator
                 continue;
             }
 
-            yield from (new LocateComposerPackageSourceFiles())->__invoke($installedPackages[$vendorName], $vendorDir);
+            yield from (new LocateComposerPackageSourceFiles(false))->__invoke($installedPackages[$vendorName], $vendorDir);
```

Comment:
Probably better to have distinct locators, rather than a configurable one (notice how the booleans are percolating through the code)

Reply:
Also where is the case for the dev files? Here a hardcoded "no-dev-files" gets passed. I'd expect the other case to be around this line somewhere, probably in another `yield`.

Reply:
> Probably better to have distinct locators, rather than a configurable one (notice how the booleans are percolating through the code)

Yes. I was thinking about this all along. More in the following comments.

Reply:
By the way, the two locators will have a lot of code in common, would you prefer inheritance or composition to deduplicate it ?